refactor/modules/data_transformer.py

```diff
@@ -1,17 +1,35 @@
 
-from typing import List
+from typing import List, Any
 from modules.event_detection import Record
 
-
+RawRecord = Any
 class DataTransformer:
-    def __init__(self) -> None:
+    def __init__(self, timestep_key) -> None:
         # umap, etc.
-        pass
+        self.record_count = 0
+        self.timestep_key = timestep_key
 
 
-    def stream_data(self, clean_data) -> List[Record]:
+    def stream_data(self, raw_records: List[RawRecord]) -> List[Record]:
         """ 
-        Takes a slice of clean data 
-        and returns input for the EventDetection
+        Takes a slice of raw records 
+        and returns records for the EventDetection
         """
-        return []
\ No newline at end of file
+        records = []
+        for raw_record in raw_records:
+            record: Record = {
+                "time": self.record_count 
+                        if self.timestep_key == "record_count" 
+                        else raw_record[self.timestep_key],
+                "categorical": {},
+                "numerical": {},
+            }
+            for key, val in raw_record.items():
+                if type(val) == int or type(val) == float:
+                    record["numerical"][key] = val
+                else:
+                    record["categorical"][key] = val
```

Comment:
This should probably change to convert categorical values to a numeric representation